platforms/software/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -115,6 +115,7 @@ void generate() {
         String jarFileRelativePath = resolver.resolveAsRelativePath(jarFileDestination);
         File propertiesFile = getPropertiesFile();
         Properties existingProperties = propertiesFile.exists() ? GUtil.loadProperties(propertiesFile) : null;
+        Integer propertyTimeout = networkTimeout.getOrElse(WrapperDefaults.NETWORK_TIMEOUT);
```

Comment:
❌  PTAL at this code:
* https://github.com/gradle/gradle/blob/2d5e045e24414b0db11a55a86762078c698bab12/platforms/software/build-init/src/main/java/org/gradle/api/tasks/wrapper/internal/WrapperGenerator.java#L98

Here, if the network timeout is set, it's written to the properties file. However, if it's not set, it's not written.

So here, you should try getting the value from the loaded properties (`existingProperties.get("networkTimeout")`) and leave it as null if it's not present there.